sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1427,4 +1427,74 @@ public void testInsertDefaultExpressions() {
                "5| cr8\n")
         );
     }
+
+    @Test
+    public void testInsertGeneratedExpressionsInNestedObjects() {
+        SQLActionException e;
+        execute("create table t (" +
+                " id int," +
+                " o object as (username text, " +
+                "               pwd text as '***', " +
+                "               nested object as (secret text as '***')" +
+                "               )" +
+                ") with (number_of_replicas=0)");
+
+        // do not provide value for nested generated expression
+        execute("insert into t (id, o) values (1, {username='crate'})");
+        execute("insert into t (id, o) select 2, {username='crate'}");
+
+        // provide 'valid' value for 1st level only nested generated expression (given == generated)
+        execute("insert into t (id, o) values (3, {username='crate', pwd='***'})");
+        execute("insert into t (id, o) select 4, {username='crate', pwd='***'}");
+
+        // provide 'valid' value for 2nd level only nested generated expression (given == generated)
+        execute("insert into t (id, o) values (5, {username='crate', nested={secret='***'}})");
+        execute("insert into t (id, o) select 6, {username='crate', nested={secret='***'}}");
+
+        // provide 'valid' values for all-level nested generated expressions (given == generated)
+        execute("insert into t (id, o) values (7, {username='crate', pwd='***', nested={secret='***'}})");
+        execute("insert into t (id, o) select 8, {username='crate', pwd='***', nested={secret='***'}}");
```

Comment:
Why do we need to test for different levels? will this add any value? If not please remove, keep the test scope low, its otherwise hard to read, understand and maintain.

Reply:
Removed, see https://github.com/crate/crate/pull/8811#discussion_r296664485